components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/TomcatCookieAuthenticationHandler.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.auth.service.handler.impl;
+
+import org.apache.catalina.connector.Request;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.auth.service.AuthenticationContext;
+import org.wso2.carbon.identity.auth.service.AuthenticationResult;
+import org.wso2.carbon.identity.auth.service.AuthenticationStatus;
+import org.wso2.carbon.identity.auth.service.handler.AuthenticationHandler;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.core.handler.InitConfig;
+import org.wso2.carbon.utils.ServerConstants;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+import static org.wso2.carbon.identity.auth.service.util.Constants.COOKIE_AUTH_HEADER;
+import static org.wso2.carbon.identity.auth.service.util.Constants.JSESSIONID;
+import static org.wso2.carbon.identity.auth.service.util.Constants.REQUEST;
+
+/**
+ * This handler is used to authenticate the rest APIs based on the set-cookie obtained from the AuthenticationAdmin
+ * Service.
+ */
+public class TomcatCookieAuthenticationHandler extends AuthenticationHandler {
+
+    private static final Log log = LogFactory.getLog(TomcatCookieAuthenticationHandler.class);
+
+    @Override
+    public void init(InitConfig initConfig) {
```

Comment:
Do we need to override this ?

Reply:
This was required since AuthenticationHandler was not extended from AbstractIdentityMessageHandler